extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts

```diff
@@ -0,0 +1,401 @@
+			// as there's no other API that'll give us this for now
+			let result = await this.waitOnMountStatusChange(controller);
+			let msg = result.state === MountState.Ready ? localize('mount.task.complete', "Mounting HDFS folder is complete")
+				: localize('mount.task.inprogress', "Mounting is likely to complete, check back later to verify");
+			op.updateStatus(azdata.TaskStatus.Succeeded, msg);
+		} catch (error) {
+			const errMsg = localize('mount.task.error', "Error mounting folder: {0}", (error instanceof Error ? error.message : error));
+			vscode.window.showErrorMessage(errMsg);
+			op.updateStatus(azdata.TaskStatus.Failed, errMsg);
+		}
+	}
+
+	private waitOnMountStatusChange(controller: ClusterController): Promise<MountInfo> {
+		return new Promise<MountInfo>((resolve, reject) => {
+			const waitTime = 5 * 1000; // 5 seconds
+			const maxRetries = 30;	// 5 x 30 = 150 seconds. After this time, can assume things are "working" as 2 min timeout passed
+			let waitOnChange = async (retries: number) => {
+				try {
+					let mountInfo = await this.getMountStatus(controller, this.props.hdfsPath);
+					if (mountInfo && mountInfo.error || mountInfo.state === MountState.Error) {
+						reject(new Error(mountInfo.error ? mountInfo.error : localize('mount.error.unknown', "Unknown error occurred during the mount process")));
+					} else if (mountInfo.state === MountState.Ready || retries <= 0) {
+						resolve(mountInfo);
+					} else {
+						setTimeout(() => {
+							waitOnChange(retries - 1).catch(e => reject(e));
+						}, waitTime);
+					}
+				} catch (err) {
+					reject(err);
+				}
+			};
+			waitOnChange(maxRetries);
+		});
+	}
+
+	private async getMountStatus(controller: ClusterController, path: string): Promise<MountInfo> {
+		let statusResponse = await controller.getMountStatus(path);
+		if (statusResponse.mount) {
+			return Array.isArray(statusResponse.mount) ? statusResponse.mount[0] : statusResponse.mount;
+		}
+		return undefined;
+	}
+}
+
+abstract class HdfsDialogBase<T extends DialogProperties> {
+
+	protected dialog: azdata.window.Dialog;
+	protected uiModelBuilder: azdata.ModelBuilder;
+
+	protected urlInputBox: azdata.InputBoxComponent;
+	protected authDropdown: azdata.DropDownComponent;
+	protected usernameInputBox: azdata.InputBoxComponent;
+	protected passwordInputBox: azdata.InputBoxComponent;
+
+	constructor(private title: string, protected model: HdfsDialogModelBase<T>) {
+	}
+
+	public showDialog(): void {
+		this.createDialog();
+		azdata.window.openDialog(this.dialog);
+	}
+
+	private createDialog(): void {
+		this.dialog = azdata.window.createModelViewDialog(this.title);
+		this.dialog.registerContent(async view => {
+			this.uiModelBuilder = view.modelBuilder;
+
+			this.urlInputBox = this.uiModelBuilder.inputBox()
+				.withProperties<azdata.InputBoxProperties>({
+					placeHolder: localize('textUrlLower', 'url'),
```

Comment:
looks like all the string values in this file are in single quotes, they should be double quotes. e.g. "url"

Reply:
Thanks - copy/paste error from existing credentials dialog... fixed